### PR TITLE
Add support for zhimi.heater.mc2

### PR DESCRIPTION
**Pull request by bafonins**
This PR adds support for Xiaomi Mi Smart Space Heater S (zhimi.heater.mc2).

References: https://github.com/rytilahti/python-miio/issues/880
miot spec: https://miot-spec.org/miot-spec-v2/instance\?type\=urn:miot-spec-v2:device:heater:0000A01A:zhimi-mc2:1

<details>
<summary>Details</summary>
Product:
Screenshot 2020-12-25 at 15 21 09

Mi Home UI:
ui1
ui2

</details>

Added unit tests as well as tested the new heater module on an actual device.

Please let me know if I missed anything.

### CODE REVIEW

File: miio/heater_miot.py [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+
+class HeaterMiotException(DeviceException):
+    pass
+
+
+class HeaterMiotStatus:
+    """Container for status reports from the heater which uses the MIoT protocol."""
+
+    def __init__(self, data: Dict[str, Any]) -> None:
+        self.data = data
```
Comment (rytilahti) [inline miio/heater_miot.py:43]: Please add an example data dictionary to the docstring, see other miot devices for example.

**Review by rytilahti — CHANGES_REQUESTED** [review]
Looks fairly straightforward to me, only a minor change needed before this can be merged. Thanks for the PR!

File: miio/heater_miot.py [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+        return self.data["power"]
+
+    @property
+    def target_temperature(self) -> int:
+        """Target temperature."""
+        return self.data["target_temperature"]
+
+    @property
+    def countdown_time(self) -> int:
+        """Countdown time."""
```
Comment (rytilahti) [inline miio/heater_miot.py:62]: What does this indicate? Time until the device gets turned off? Maybe the property naming should follow the same format as in heater?

File: miio/heater_miot.py [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+                self.buzzer,
+                self.child_lock,
+                self.countdown_time,
+            )
+        )
+        return s
+
+
+class HeaterMiot(MiotDevice):
+    """Main class representing the heater which uses MIoT protocol."""
```
Comment (rytilahti) [inline miio/heater_miot.py:108]: Please add the model also to this docstring to let readers to know which device(s) are supported.

**Review by rytilahti — CHANGES_REQUESTED** [review]

File: miio/heater_miot.py (hunk 1 above)
Comment (rytilahti) [reply miio/heater_miot.py:43]: Also, please update the README.md to add this to supported devices.

File: miio/heater_miot.py (hunk 2 above)
Comment (bafonins) [reply miio/heater_miot.py:75]: Exactly. The `countdown_time` field defines the number hours before the device goes off. 
The problem is here that the spec defines this field as `countdown_time` while for miio heater it is `delay_off_countdown`. Shall overwrite it to match the field name in heater?

**Comment by bafonins** [comment]
@rytilahti fixed according to your comments. let me know if I missed anything

File: miio/heater_miot.py (hunk 2 above)
Comment (rytilahti) [reply miio/heater_miot.py:75]: Okay, looking at the current code base, there are several instances of `delay_off_countdown` which returns the value in seconds.

So I'd say yes, it's better use that same naming scheme (and return the value in seconds even when the precision is much lower) just to be consistent.

File: miio/heater_miot.py [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+        """True if child lock is on, False otherwise."""
+        return self.data["child_lock"] is True
+
+    @property
+    def buzzer(self) -> bool:
+        """True if buzzer is turned on, False otherwise."""
+        return self.data["buzzer"] is True
+
+    @property
+    def brightness(self) -> LedBrightness:
```
Comment (rytilahti) [inline miio/heater_miot.py:94]:
```suggestion
    def led_brightness(self) -> LedBrightness:
```

File: miio/heater_miot.py [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+                self.buzzer,
+                self.child_lock,
+                self.delay_off_countdown,
+            )
+        )
+        return s
+
+
+class HeaterMiot(MiotDevice):
+    """Main class representing the Xiaomi Smart Space Heater S."""
```
Comment (rytilahti) [inline miio/heater_miot.py:126]:
```suggestion
    """Main class representing the Xiaomi Smart Space Heater S (zhimi.heater.mc2)."""
```
Please add a note in this docstring about the supported model. 

At some point in the future this project will hopefully have a model<->implementation class type of mapping for automatic generation of supported devices in docs, discovery helper and potential other uses. When the time comes, having this information in the source will make the process much easier.

**Review by rytilahti — APPROVED** [review]
One last very minor change and this is good to go. Thanks for the PR! :1st_place_medal:

**Comment by ee02217** [comment]
This is great! Thanks!
How about adding support to Home Assistant integration? 👍

**Pull request merged**